activesupport/lib/active_support/json_with_marshal_fallback.rb

```diff
@@ -0,0 +1,25 @@
+module ActiveSupport
+  class JsonWithMarshalFallback
+
+    cattr_accessor :fallback_to_marshal_serialization, instance_accessor: false, default: true
+    cattr_accessor :use_marshal_serialization, instance_accessor: false, default: true
+
+    def self.dump(value)
+      if self.use_marshal_serialization
+        Marshal.dump(value)
+      else
+        JSON.encode(value)
+      end
+    end
+
+    def self.load(value)
+      JSON.decode(value)
```

Comment:
Yes, do this please.

Reply:
Why go the raise/rescue route at all? We know marshal data will start with MARSHAL_SIGNATURE, and we know that JSON data won't. I think this can just be an `if`?

Reply:
You're right thanks for the feedback! I've tested with just the conditional locally and it works as intended, going to push that up.

Reply:
Below is the new approach. It keeps the `raise` that attempts to prevent binary data from being passed to the JSON decoder when the fallback is turned off.

```ruby
      def load(value)
        if self.fallback_to_marshal_deserialization
          if value.starts_with?(MARSHAL_SIGNATURE)
            logger.warn("JsonWithMarshalFallback: Marshal load fallback occurred.") if logger
            Marshal.load(value)
          else
            JSON.decode(value)
          end
        else
          raise ::JSON::ParserError if value.start_with?(MARSHAL_SIGNATURE)
          JSON.decode(value)
        end
      end

```